dace/sdfg/validation.py

```diff
@@ -191,6 +191,22 @@ def validate_state(state: 'dace.sdfg.SDFGState',
         except Exception as ex:
             raise InvalidSDFGNodeError("Node validation failed: " + str(ex), sdfg, state_id, nid) from ex
 
+        if Config.get_bool('check_race_conditions'):
+            if isinstance(node, nd.AccessNode):
+                incoming_edge_memlet_ranges = []
+                for e in state.in_edges(node):
+                    memlet_range = str(e.data.subset).split(':')
+                    incoming_edge_memlet_ranges.append([int(memlet_range[0]), int(memlet_range[1])])
```

Comment:
this will not work for symbolic ranges (e.g., `0:N`)

Reply:
I tried using sympy's `compare` method for this purpose, do you think it is acceptable?
(Ref : https://docs.sympy.org/latest/modules/core.html#sympy.core.basic.Basic.compare)